### PR TITLE
Convert provision plugins' step data to our field implementation

### PR DESCRIPTION
**Pull request by happz**
This unites options with plugin *and* guest data, and with inheritance, we also get some options shared among plugins through the step data instead of overriden `options()` methods.

### CODE REVIEW

**Comment by happz** [comment]
Blocked because of `field()` lacking the support for envvars.

File: tmt/steps/provision/__init__.py [hunk 1]
```diff
@@ -848,17 +848,37 @@ class GuestSshData(GuestData):
...
+        default=None,
+        option=('-p', '--password'),
+        metavar='PASSWORD',
+        help='Password for login into the guest system.')
+    ssh_option: List[str] = field(
+        default_factory=list,
+        option='--ssh-option',
+        metavar="OPTION",
+        multiple=True,
+        help="Specify additional SSH option. "
```
Comment (psss) [inline tmt/steps/provision/__init__.py:878]:
```suggestion
        help="Specify an additional SSH option. "
```

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for the cleanup! Just a few minor suggestions.

File: tmt/utils.py [hunk 2]
```diff
@@ -4350,6 +4350,66 @@ def dataclass_normalize_field(
...
+def normalize_storage_size(
+        key_address: str,
+        value: Any,
+        logger: tmt.log.Logger) -> int:
+    """
+    Normalize a storage size.
+
+    As of now, it's just a simple integer with units interpreted by the owning
+    plugin. In the future, we want this function to switch to proper units
+    and return ``ping.Quantity`` instead.
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/utils.py]: Fixed in 2895cbbb.

File: tmt/utils.py [hunk 3]
```diff
@@ -4350,6 +4350,66 @@ def dataclass_normalize_field(
...
+
+
+def normalize_optional_int(
+        key_address: str,
+        value: Optional[str],
+        logger: tmt.log.Logger) -> Optional[int]:
+    """
+    Normalize an integer that may be unset as well.
+
+    For a field that takes an integer input, but might be also left out,
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/utils.py:4381]: Fixed in 2895cbbb. It turns out that normalization callbacks should probably accept `Any` as their input, and deal with types in their body.

I'm not sure whether it's necessary to mention the `str` coming from Click, strings are everywhere :)

File: tmt/steps/provision/__init__.py (hunk 1 above)
Comment (happz) [reply tmt/steps/provision/__init__.py]: Fixed in 2895cbbb.

**Review by psss — APPROVED** [review]

**Pull request merged**